### ISSUE TITLE
Prepare, test, and release v1.0.2

### ISSUE THREAD

**Issue by SebastianZimmeck** [privacy-tech-lab/privacyflash-pro#17]
@rgoldstein01, can you prepare, test and release a new version? Here are instructions that @davebaraka provided for the previous release:

I packaged the generator (attached below). I am running Mac OS Version 10.15 (19A583).

I was able to package the generator by first creating and activating a python virtual environment. Then, installing the dependencies for the generator in the virtual environment as well as Pyinstaller. To package the generator, run `python3 -m eel app.py interface --onefile --noconsole --add-data "data:data"` in the `policygenerator` directory with all the python files.  The packaged generator should appear in a `/dist` folder.

Packaging the generator may still work without the virtual environment, but not recommended, at least for production.

An `app.spec` file will also appear in the `policygenerator` directory. This file contains the configuration settings for PyInstaller to package the python script. So, running `python3 -m PyInstaller app.spec` will also appropriately package the generator. You can modify this file to change the name of the packaged script, add an icon, etc. [Spec Files](https://pythonhosted.org/PyInstaller/spec-files.html)

_Originally posted by @davebaraka in https://github.com/privacy-tech-lab/privacyflash-pro-dev/issues/10#issuecomment-541496215_

**Comment by SebastianZimmeck** [comment]
Some further info @davebaraka provided:

Some notes for myself:

`python3 -m PyInstaller app.spec `

```
# -*- mode: python ; coding: utf-8 -*-

block_cipher = None


a = Analysis(['app.py'],
             pathex=['/Users/davebaraka/Development/privacyflash-pro-dev/policygenerator'],
             binaries=[],
             datas=[('/Users/davebaraka/Development/privacyflash-pro-dev/policygenerator/env/lib/python3.7/site-packages/eel/eel.js', 'eel'), ('interface', 'interface'), ('spec', 'spec')],
             hiddenimports=['bottle_websocket'],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher,
             noarchive=False)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          a.binaries,
          a.zipfiles,
          a.datas,
          [],
          name='PrivacyFlash Pro',
          debug=False,
          bootloader_ignore_signals=False,
          strip=False,
          upx=True,
          upx_exclude=[],
          runtime_tmpdir=None,
          console=False )
app = BUNDLE(exe,
             name='PrivacyFlash Pro.app',
             icon='./interface/img/icon.icns',
             bundle_identifier=None,
             info_plist={
            'CFBundleShortVersionString': '1.0.1',
            },
            )

```

_Originally posted by @davebaraka in https://github.com/privacy-tech-lab/privacyflash-pro-dev/issues/10#issuecomment-570984325_

(cc'ing @kalicki1)

**Comment by rgoldstein01** [comment]
I am currently following the directions @davebaraka has put and I have successfully created a package without errors. However, when I run the app, everything runs smoothly until I analyze an app and then does not go to the next step. The virtual environment is working well because when I run python3 app.py in the virtual env it works but once it is packaged it gets stopped there. I am trying to find an error log but I am unable to find one.
Here is the app if you'd like to try

[PrivacyFlash Pro.zip](https://github.com/privacy-tech-lab/privacyflash-pro-dev/files/4704923/PrivacyFlash.Pro.zip)

**Comment by SebastianZimmeck** [comment]
There seems to be something wrong with the packaging, indeed. I am getting the same result as @rgoldstein01. When running with `python3 app.py` (without virtual environment), everything works. But the packaged version never gets to the results screen. (I am using macOS Catalina 10.15.5).

@kalicki1, if you have time, please take a look at this as well.

**Comment by SebastianZimmeck** [comment]
As @rgoldstein01 mentioned in today's call, @davebaraka's [instructions](https://github.com/privacy-tech-lab/privacyflash-pro/issues/17#issue-627351524) need to be slightly updated as the `data` directory in our repo was renamed to `spec`.

**Comment by SebastianZimmeck** [comment]
Any progress here?

**Comment by davebaraka** [comment]
The generator has been packaged and it has been added as a pre-release to 'releases'. If everything seems fine, we can make it a release.

The problem was that the file structure was changed when unit testing was added (we also shifted to using python modules rather than scripts) and determining the application path when packaged vs in development is different. In other words, the generator could not find the `spec` folder when packaged. So I used `sys._MEIPASS` as described [here](https://stackoverflow.com/a/42615559).

My instructions to package the generator should still work.
